integration/nwo/token/fabric/cc/tcc.go

```diff
@@ -17,20 +17,139 @@ import (
+	"github.com/hyperledger-labs/fabric-token-sdk/integration/nwo/token/fabric"
 	topology3 "github.com/hyperledger-labs/fabric-token-sdk/integration/nwo/token/topology"
 	. "github.com/onsi/gomega"
 )
 
+const (
+	DefaultTokenChaincode                    = "github.com/hyperledger-labs/fabric-token-sdk/token/services/network/fabric/tcc/main"
+	DefaultTokenChaincodeParamsReplaceSuffix = "/token/services/network/fabric/tcc/params.go"
+)
+
+var logger = flogging.MustGetLogger("token-sdk.integration.token.fabric.cc")
+
+type fabricPlatform interface {
+	Topology() *topology.Topology
+	UpdateChaincode(name string, version string, path string, file string)
+}
+
 type TCC struct {
 	Chaincode *topology.ChannelChaincode
 }
 
-func (p *NetworkHandler) tccSetup(tms *topology3.TMS, cc *topology.ChannelChaincode) (*topology.ChannelChaincode, uint16) {
+type GenericBackend struct {
+	TokenChaincodePath                string
+	TokenChaincodeParamsReplaceSuffix string
+	TokenPlatform                     common2.TokenPlatform
+}
+
+func NewDefaultGenericBackend(tokenPlatform common2.TokenPlatform) *GenericBackend {
+	return NewGenericBackend(
+		DefaultTokenChaincode,
+		DefaultTokenChaincodeParamsReplaceSuffix,
+		tokenPlatform,
+	)
+}
+
+func NewGenericBackend(tokenChaincodePath string, tokenChaincodeParamsReplaceSuffix string, tokenPlatform common2.TokenPlatform) *GenericBackend {
+	return &GenericBackend{TokenChaincodePath: tokenChaincodePath, TokenChaincodeParamsReplaceSuffix: tokenChaincodeParamsReplaceSuffix, TokenPlatform: tokenPlatform}
+}
+
+func (p *GenericBackend) PrepareNamespace(entry *fabric.Entry, tms *topology3.TMS) {
+	orgs := tms.BackendParams["fabric.orgs"].([]string)
+
+	// Standard Chaincode
+	policy := "AND ("
+	for i, org := range orgs {
+		if i > 0 {
+			policy += ","
+		}
+		policy += "'" + org + "MSP.member'"
+	}
+	policy += ")"
+
+	var peers []string
+	for _, org := range orgs {
+		for _, peer := range p.Fabric(tms).Topology().Peers {
+			if peer.Organization == org {
+				peers = append(peers, peer.Name)
+			}
+		}
+	}
+
+	cc, _ := p.tccSetup(tms, &topology.ChannelChaincode{
+		Chaincode: topology.Chaincode{
+			Name:            tms.Namespace,
+			Version:         "Version-0.0",
+			Sequence:        "1",
+			InitRequired:    true,
+			Path:            p.TokenChaincodePath,
+			Lang:            "golang",
+			Label:           tms.Namespace,
+			Policy:          policy,
+			SignaturePolicy: policy,
+		},
+		Channel: tms.Channel,
+		Peers:   peers,
+	})
+	p.Fabric(tms).Topology().AddChaincode(cc)
+}
+
+func (p *GenericBackend) UpdatePublicParams(tms *topology3.TMS, ppRaw []byte) {
+	var cc *topology.ChannelChaincode
+	for _, chaincode := range p.Fabric(tms).Topology().Chaincodes {
+		if chaincode.Chaincode.Name == tms.Namespace {
+			cc = chaincode
+			break
+		}
+	}
+	Expect(cc).NotTo(BeNil(), "failed to find chaincode [%s]", tms.Namespace)
+
+	packageDir := filepath.Join(
```

Comment:
Maybe we can extract 114-145 to a method

```
func (p *GenericBackend) packageChaincode(tmsID TMSID, cc *topology.ChannelChaincode, chaincodeNameSuffix string) (filename, error)
```

Reply:
let's address this later